shop/shop/src/main/java/ro/msg/learning/shop/strategy/StrategyContext.java

```diff
@@ -0,0 +1,24 @@
+package ro.msg.learning.shop.strategy;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class StrategyContext {
+    @Value("${strategy}")
+    String strategy;
+    @Bean
+    public IStrategy getStrategy(){
```

Comment:
Almost perfect implementation. What more could have been done here would have been to have an `enum` with the strategies and based on it to return the correct instance. This way is also more **open-closed**